examples/tls_boring_dynamic_certs.rs

```diff
@@ -0,0 +1,220 @@
+        .await
+        .expect("graceful shutdown");
+}
+
+struct DynamicIssuer {
+    example_data: ServerAuthData,
+    second_example_data: ServerAuthData,
+    default_data: ServerAuthData,
+}
+
+impl DynamicIssuer {
+    fn new() -> Self {
+        Self {
+            example_data: example_self_signed_auth().expect("load example data"),
+            second_example_data: second_example_self_signed_auth()
+                .expect("load second example data"),
+            default_data: example_self_signed_auth().expect("load default data"),
+        }
+    }
+}
+
+impl DynamicCertIssuer for DynamicIssuer {
+    async fn issue_cert(
+        &self,
+        client_hello: ClientHello,
+        _server_name: Option<Host>,
+    ) -> Result<ServerAuthData, OpaqueError> {
+        match client_hello.ext_server_name() {
+            Some(host) => match host {
+                rama_net::address::Host::Name(domain) => {
+                    if domain == &Domain::from_static("example") {
+                        return Ok(self.example_data.clone());
+                    } else if domain == &Domain::from_static("second.example") {
+                        return Ok(self.second_example_data.clone());
+                    }
+                    Ok(self.example_data.clone())
+                }
+                rama_net::address::Host::Address(_ip_addr) => Ok(self.default_data.clone()),
+            },
+            None => Ok(self.example_data.clone()),
+        }
+    }
+}
+
+pub fn example_self_signed_auth() -> Result<ServerAuthData, OpaqueError> {
+    Ok(ServerAuthData {
+        private_key: DataEncoding::Pem(
+            std::str::from_utf8(include_bytes!("./assets/example.com.key"))
+                .expect("should decode")
+                .try_into()
+                .expect("should work"),
+        ),
+        cert_chain: DataEncoding::Pem(
+            std::str::from_utf8(include_bytes!("./assets/example.com.crt"))
+                .expect("should decode")
+                .try_into()
+                .expect("should work"),
+        ),
+        ocsp: None,
+    })
+}
+
+pub fn second_example_self_signed_auth() -> Result<ServerAuthData, OpaqueError> {
+    Ok(ServerAuthData {
+        private_key: DataEncoding::Pem(
+            std::str::from_utf8(include_bytes!("./assets/second_example.com.key"))
+                .expect("should decode")
+                .try_into()
+                .expect("should work"),
+        ),
+        cert_chain: DataEncoding::Pem(
+            std::str::from_utf8(include_bytes!("./assets/second_example.com.crt"))
+                .expect("should decode")
+                .try_into()
+                .expect("should work"),
+        ),
+        ocsp: None,
+    })
+}
+
+async fn http_service<S>(_ctx: Context<S>, _request: Request) -> Result<Response, Infallible> {
+    Ok(Response::new(
+        format!(
+            "hello client, you were served by boring tls terminator proxy issuing a dynamic certificate\r\n"
```

Comment:
Pretty certain you can just call into_response() on the &'static str. Why the newlines btw?